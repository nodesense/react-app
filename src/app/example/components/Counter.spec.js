import React from 'react';
import { shallow, mount } from 'enzyme';
import Counter from './Counter';

describe ("Counter specs ", () => {

    // smoke test - test if it render
    it('it should render', () => {
        shallow(<Counter />);
    });
    
    // snapshot test - test the final HTML
    it('it should render the expected HTML', () => {
        expect(
        mount(<Counter />).html()
        ).toMatchSnapshot();
    });
    
    // unit test - test functionality
    it('it should update the state and change the DOM after each click', () => {
        const wrapper = shallow(<Counter />);
    
        expect(
        wrapper.find('strong').text()
        ).toBe('0');
    
        wrapper.find('div').simulate('click');
    
        expect(
        wrapper.find('strong').text()
        ).toBe('1');
    });
})