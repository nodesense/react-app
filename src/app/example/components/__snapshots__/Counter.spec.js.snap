// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter specs  it should render the expected HTML 1`] = `"<div>Clicked <strong>0</strong> times!</div>"`;
